Plutarch/Repr/Data.hs

```diff
@@ -0,0 +1,209 @@
+{-# LANGUAGE FlexibleInstances #-}
```

Comment:
Needs `@since`s on publically-exposed bits, more clear explanations what everything is, and non-exposed things should be in a ``-- Helpers` subsection.